cmd/block-generator/generator/generate.go

```diff
@@ -450,33 +451,44 @@ func (g *generator) generatePaymentTxn(round uint64, intra uint64) (transactions
 	return g.generatePaymentTxnInternal(selection.(TxTypeID), round, intra)
 }
 
+const minBal = uint64(100000)
+
 func (g *generator) generatePaymentTxnInternal(selection TxTypeID, round uint64, intra uint64) (transactions.SignedTxn, transactions.ApplyData, error) {
 	defer g.recordData(track(selection))
 
+	// amounts
+	amount := uint64(1)
+	fee := uint64(1000)
+	total := amount + fee
+
+	// Select a receiver
 	var receiveIndex uint64
 	switch selection {
 	case paymentTx:
 		receiveIndex = rand.Uint64() % g.numAccounts
 	case paymentAcctCreateTx:
+		// give new accounts extra algos for sending other transactions (optin/optout)
+		amount = minBal * 100
 		g.balances = append(g.balances, 0)
-		g.numAccounts++
-		receiveIndex = g.numAccounts - 1
+		receiveIndex = g.numAccounts
+		// increment at the end in case it needs to be referenced later.
+		defer func() {
+			g.numAccounts++
+		}()
 	}
 
-	// Always send from a genesis account.
-	sendIndex := g.numPayments % g.config.NumGenesisAccounts
+	// Select a sender from genesis account
+	sendIndex := (g.numPayments + g.paymentOffset) % g.config.NumGenesisAccounts
+	// if the genesis account has insufficient balance... start checking others
```

Comment:
this is unnecessary now, right?

Reply:
Depending on the configuration this could still be needed. Not quite as obvious as killing the program, but the print statement still makes it pretty clear when you're using this fallback.

The checked-in scenarios shouldn't have a problem even for very long tests.

Reply:
I'm not sure what configurations you are referring to. There should be enough microalgos for any desirable configuration.